src/Models/RemoteExtension.ts

```diff
@@ -25,12 +25,10 @@ export class RemoteExtension {
     return await checkExtensionAvailable(ExtensionName.Remote);
   }
 
-  static async checkRemoteExtension(): Promise<void> {
+  static async checkRemoteExtension(operation: string): Promise<void> {
```

Comment:
nit: can we manage the "operation" as enum type here?
It can provide us an overview of what kind of operations will need to check remote extension. Another benefit to use enum type here is improve the re-usability for the same operation used in multiple places.

Reply:
The `Operation: string` is used when in error message to help know what operation throws error. The value of operation varies a lot through the whole project. This is why I didn't set it as enum at the very beginning.

If we have a enum type for all operations, the list will be very large. 
E.g. "create function application", "get credential from subscription id", "crete iot containerized project", "open project in remote", "configure project environment", "generate CRC", etc. 
Really a big list since operations varies.